collector/src/compile/execute/profiler.rs

```diff
@@ -51,6 +52,68 @@ impl Profiler {
                 | Profiler::DepGraph
         )
     }
+
+    pub fn diff_prereqs(&self) -> anyhow::Result<()> {
```

Comment:
Can we just move this check directly into `cachegrind_diff`? I don't think that it's worth it to create a separate interface on Profiler for this.